tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java

```diff
@@ -0,0 +1,697 @@
+      blockingQueue.setMaxSizeBytes(100 * 1024)
+         .setAddressFullMessagePolicy(AddressFullMessagePolicy.FAIL)
+         .setSlowConsumerPolicy(SlowConsumerPolicy.KILL).setSlowConsumerThreshold(0).setSlowConsumerCheckPeriod(TimeUnit.MILLISECONDS.toSeconds(opTimeoutMillis))
+         .setAutoDeleteQueues(false).setAutoDeleteAddresses(false); // so slow consumer can kick in!
+
+      Configuration baseConfig = new ConfigurationImpl();
+      baseConfig.getAddressesSettings().put(qName, blockingQueue);
+
+      BrokerBalancerConfiguration balancerConfiguration = new BrokerBalancerConfiguration();
+      balancerConfiguration.setName(balancerConfigName).setTargetKey(TargetKey.ROLE_NAME).setTargetKeyFilter("(?<=^EQ_).*"); // strip EQ_ prefix
+      baseConfig.addBalancerConfiguration(balancerConfiguration);
+
+      // prepare two nodes
+      for (int nodeId = 0; nodeId < 2; nodeId++) {
+         Configuration configuration = baseConfig.copy();
+         configuration.setName("Node" + nodeId);
+         configuration.setBrokerInstance(new File(getTestDirfile(), configuration.getName()));
+         configuration.addAcceptorConfiguration("tcp", "tcp://localhost:" + (base_port + (nodeId)) + "?redirect-to=" + balancerConfigName + ";amqpCredits=" + credit + ";amqpMinCredits=" + creditMin);
+         nodes.add(new EmbeddedActiveMQ().setConfiguration(configuration));
+         nodes.get(nodeId).setSecurityManager(customSecurityManager);
+         nodes.get(nodeId).setMbeanServer(mBeanServer);
+      }
+
+      // node0 initially handles both producer & consumer (head & tail)
+      nodes.get(0).getConfiguration().getBalancerConfigurations().get(0).setLocalTargetFilter("PRODUCER|CONSUMER");
+      nodes.get(0).start();
+   }
+
+   @Test (timeout = 20000)
+   public void testScale0_1() throws Exception {
+
+      prepareNodesAndStartCombinedHeadTail();
+
+      // slow consumer, delay on each message received
+      EQConsumer eqConsumer = new EQConsumer(urlForNodes(nodes));
+      eqConsumer.start();
+
+      // verify consumer reconnects on no messages
+      assertTrue(Wait.waitFor(() -> eqConsumer.connectionCount.get() > 1));
+
+      EQProducer eqProducer = new EQProducer(urlForNodes(nodes));
+      eqProducer.start();
+
+      // verify producer reconnects on fail full!
+      assertTrue(Wait.waitFor(() -> eqProducer.connectionCount.get() > 1));
+
+      // operator mode, poll queue control - to allow producer to continue, activate next broker in the 'chain'
+      AddressControl addressControl0 = (AddressControl) ManagementControlHelper.createProxy(node0NameBuilder.getAddressObjectName(qNameSimple), AddressControl.class, mBeanServer);
+
+      AddressControl finalAddressControl = addressControl0;
+      assertTrue(Wait.waitFor(() -> {
+         int usage = finalAddressControl.getAddressLimitPercent();
+         System.out.println("Node0 usage % " + usage);
+         return usage == 100;
```

Comment:
I guess usage percent could never be 100

Reply:
good catch, the test did indeed need some work with the fixed getAddressLimitPercent :-)